nucleus/src/main/java/za/co/absa/subatomic/infrastructure/team/TeamAutomationHandler.java

```diff
@@ -59,28 +59,51 @@ public void createNewTeam(TeamEntity newTeamEntity) {
             teamSlackIdentity = new TeamSlackIdentity(
                     newTeamEntity.getSlackDetails().getTeamChannel());
         }
-        TeamCreated teamCreated = new TeamCreated(newTeamEntity.getTeamId(),
+
+        TeamCreated teamCreated = new TeamCreated (newTeamEntity.getTeamId(),
                 newTeamEntity.getName(), newTeamEntity.getDescription(),
                 new TeamMemberId(newTeamEntity.getCreatedBy().getMemberId()),
                 teamSlackIdentity);
 
+        AtomistMemberBase atomistMemberBase = new AtomistMemberBase(
+                teamMemberEntity.getFirstName(),
+                teamMemberEntity.getDomainUsername(),
+                teamMemberSlackIdentity);
+
         TeamCreatedWithDetails newTeam = new TeamCreatedWithDetails(teamCreated,
-                new AtomistMemberBase(
-                        teamMemberEntity.getFirstName(),
-                        teamMemberEntity.getDomainUsername(),
-                        teamMemberSlackIdentity));
+                atomistMemberBase);
 
         ResponseEntity<String> response = restTemplate.postForEntity(
                 atomistConfigurationProperties.getTeamCreatedEventUrl(),
                 newTeam,
                 String.class);
 
         if (response.getStatusCode().is2xxSuccessful()) {
-            log.info("Atomist has ingested event successfully: {} -> {}",
+            log.info("Atomist has ingested TeamCreatedEvent successfully: {} -> {}",
                     response.getHeaders(), response.getBody());
         }
     }
 
+    public void teamSlackChannelCreated(AtomistTeam atomistTeam, AtomistMemberBase atomistMemberBase) {
```

Comment:
This should take in Entity classes the same way the other functions in this class do. Otherwise we are introducing too much cross pollination between the adapter and infrastructure packages.